### PR TITLE
Move the Cohorts (index/show) pages under admin namespace

### PR DESCRIPTION
**Pull request by Natblow**
## What's the change?
With @Caleb-Mitchell we moved the cohort index/show into an admin interface.

## What key workflows are impacted?
Routes.
## Cleanup
We had to rename all the models/views/controllers into namespace, as well as in the Specs
## Demo / Screenshots
image

## Issue ticket number and link
#416 

## Checklist before requesting a review

- [X] Did you add appropriate automated tests?
- [X] Is everything elevated to the highest level? (e.g., can logic be moved out of view into controllers, or out of controllers into models?)
- [X] Is there any linting that needs to be executed?

### CODE REVIEW

File: app/models/admin/user_mentee_application_cohort.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #  created_at        :datetime         not null
 #  updated_at        :datetime         not null
 #
-class UserMenteeApplicationCohort < ApplicationRecord
+class Admin::UserMenteeApplicationCohort < ApplicationRecord
```
Comment (toyhammered) [inline app/models/admin/user_mentee_application_cohort.rb:12]: So we shouldn't be changing the model namespace. This should remain `UserMenteeApplicationCohort` still.

**Review by toyhammered — CHANGES_REQUESTED** [review]
This seems mostly fine minus moving the model (which kinda cascades removing some of the changes you did).

I feel like this is also partially my fault for the lack of clarity, but generally you only need to scope your controllers, not the models (multiple controllers can talk to the same model)

**Comment by Caleb-Mitchell** [comment]
Great we corrected now. 

We encountered an issue on `app/views/admin/user_mentee_application_cohorts/_table_row` where we needed to update the path method. That's why we originally tried namespacing the model and it worked. 

But, actually we just need to write it like this `admin_user_mentee_application_cohort_path(cohort)` :smile: .

File: app/controllers/user_mentee_applications_controller.rb [hunk 2]
```diff
@@ -67,7 +67,7 @@ def update; end
   private
 
   def set_active_cohort
-    @active_cohort = UserMenteeApplicationCohort.active
+    @active_cohort = Admin::UserMenteeApplicationCohort.active
```
Comment (toyhammered) [inline app/controllers/user_mentee_applications_controller.rb:70]: This shouldn't be namespaced

File: db/seeds.rb [hunk 3]
```diff
@@ -89,12 +89,12 @@ def generate_pair_request_data(users:, status:, pair_date: nil)
...
 
   # Current upcoming cohort
-  active_cohort = UserMenteeApplicationCohort.create!(
+  active_cohort = Admin::UserMenteeApplicationCohort.create!(
     name: 'Fall 2023',
     active_date_range: Time.zone.today..3.months.from_now
   )
 
-  original_cohort = UserMenteeApplicationCohort.create!(
+  original_cohort = Admin::UserMenteeApplicationCohort.create!(
```
Comment (toyhammered) [inline db/seeds.rb:97]: These shouldn't either be namespaced.

File: lib/tasks/reapplication_notification_task.rake [hunk 4]
```diff
@@ -1,11 +1,11 @@
 class NoActiveCohortError < StandardError; end
 
 task enqueue_reapplication_notifications: :environment do
-  active_cohort = UserMenteeApplicationCohort.active
+  active_cohort = Admin::UserMenteeApplicationCohort.active
```
Comment (toyhammered) [inline lib/tasks/reapplication_notification_task.rake:4]: Also these

File: spec/mailers/previews/mentee_application_mailer_preview.rb [hunk 5]
```diff
@@ -29,7 +29,7 @@ def notify_applicant_of_code_challenge_approved
 
   def notify_applicant_to_reapply
     MenteeApplicationMailer.with(recipient: User.last,
-      active_cohort_name: UserMenteeApplicationCohort.active.name).notify_applicant_to_reapply
+      active_cohort_name: Admin::UserMenteeApplicationCohort.active.name).notify_applicant_to_reapply
```
Comment (toyhammered) [inline spec/mailers/previews/mentee_application_mailer_preview.rb:32]: and this

**Review by toyhammered — CHANGES_REQUESTED** [review]
Still a few more lingering `Admin::` that shouldn't be there.

**Comment by Natblow** [comment]
Thank you! Done now.

**Review by toyhammered — APPROVED** [review]

**Pull request merged**